frontend/packages/dev-console/src/components/import/DeployImage.tsx

```diff
@@ -73,24 +78,38 @@ const DeployImage: React.FC<DeployImageProps> = ({ namespace }) => {
   };
 
   const handleSubmit = (values, actions) => {
-    const {
-      project: { name: projectName },
-    } = values;
-
-    const dryRunRequests: Promise<K8sResourceKind[]> = createResources(values, true);
-    dryRunRequests
-      .then(() => {
-        const requests: Promise<K8sResourceKind[]> = createResources(values);
-        return requests;
-      })
-      .then(() => {
-        actions.setSubmitting(false);
-        history.push(`/topology/ns/${projectName}`);
-      })
-      .catch((err) => {
-        actions.setSubmitting(false);
-        actions.setStatus({ submitError: err.message });
-      });
+    submitAction(values, actions);
+    // const {
+    //   project: { name: projectName },
+    //   name,
+    //   isi: { name: isiName, tag },
+    // } = values;
+    // if (values.serverless && values.serverless.trigger) {
+    //   createKnativeService(name, projectName, isiName, tag)
+    //     .then(() => {
+    //       actions.setSubmitting(false);
+    //       history.push(`/topology/ns/${projectName}`);
+    //     })
+    //     .catch((err) => {
+    //       actions.setSubmitting(false);
+    //       actions.setStatus({ submitError: err.message });
+    //     });
+    // } else {
+    //   const dryRunRequests: Promise<K8sResourceKind[]> = createResources(values, true);
+    //   dryRunRequests
+    //     .then(() => {
+    //       const requests: Promise<K8sResourceKind[]> = createResources(values);
+    //       return requests;
+    //     })
+    //     .then(() => {
+    //       actions.setSubmitting(false);
+    //       history.push(`/topology/ns/${projectName}`);
+    //     })
+    //     .catch((err) => {
+    //       actions.setSubmitting(false);
+    //       actions.setStatus({ submitError: err.message });
+    //     });
+    // }
```

Comment:
@christianvogt you asked for this to moved on the other PR that copies this one. So I moved it, but I don't see the value. 
I left it commented out so you could see what was moved and to turn it back on as needed. 
The serverless function does not support the dryness test so they need to be separated. This also takes the `action`. I think it makes sense to keep it here but I've moved it to see if this is really what you want.

Reply:
I thought dry run was for any resource?
Regardless, I was not asking to move the entire function, but rather the if conditions related to if serverless or not into the util so that the Component only needs to deal with whether the creation failed or succeeded and then perform the necessary actions. The util would be responsible for creating the resources whether they be knative or not.

Reply:
I think when I first put this together I had issues with the dry run and so had to move it out. That is the reason it is separated like you see here. 
I'll take another look and confirm it.
If it can't work with dry run then I'm thinking to put it all back the way it was.

Reply:
`dry run` doesn't seem to work with Knative, it returns `5XX`

Reply:
I worked out how to move it to the submit utils.